src/basis_app_class.py

```diff
@@ -0,0 +1,107 @@
+from dataclasses import InitVar, dataclass, field
+from PyQt5.QtWidgets import QApplication, QStackedWidget
+from .singleton import Singleton
+
+@dataclass
+class PhoneApp(metaclass=Singleton):
+    """Singleton class which is basis for the entire app.
+
+    Params:
+        master (QApplication): main component of PyQt5 app, the __str__ method doesn't contain this variable
+        title (str): title of PhoneApp application
+        style_file (str): name of css file which contains all stylesheets of widgets
+        first_window (object): class object which is shown when app is turned on
+        height (int): height of the window, property which is initialized during the __post_init__ method
+        width (int): width of the window, property which is initialized during the __post_init__ method
+        widget (QStackedWidget): component which will contain every view of application, the __str__ method doesn't contain this variable and user cannot initialize it
+        _windows (dict): initially empty dictionary which will contain every class as key and every instance of this class as a value, the __str__ method doesn't contain this variable and user cannot initialize it"""
+    master: QApplication = field(repr=False)
+    title: str
+    style_file: str
+    first_window: object
+    height: InitVar[int]
+    width: InitVar[int]
+    widget: QStackedWidget = field(init=False, repr=False)
+    _windows: dict = field(init=False, repr=False)
+
+    def __post_init__(self, screen_widht: int, screen_height: int) -> None:
+        """Initialize a new PhoneApp instance
+
+        Params:
+            screen_widht (int): width of app
+            screen_height (int): height of app"""
+        self.widget = QStackedWidget()
+        self.master.setStyleSheet(open(f'src/{self.style_file}').read())
```

Comment:
I know that Python close files automatically, but please close this file :)
I think it's more clear and in a good pratice